nixos/modules/services/web-apps/ackee.nix

```diff
@@ -0,0 +1,135 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ackee;
+  settings = cfg.defaultSettings // cfg.settings;
+in
+{
+  options.services.ackee = {
+    enable = mkEnableOption "the Ackee analytics tool";
+
+    defaultSettings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ str path package ]));
+      default = {
+        ACKEE_MONGODB = "mongodb://localhost:27017/ackee";
+        ACKEE_USERNAME = "ackee";
+        ACKEE_PORT = "3000";
+      };
+      example = literalExample "{}";
+      description = ''
+        Settings can be overwritten in <option>settings</option>.
+        To remove all default settings, set <code>defaultSettings = {}</code>.
+      '';
+    };
+
+    settings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ str path package ]));
+      default = { };
+      example = literalExample ''
+        {
+          ACKEE_MONGODB = "mongodb://localhost:27017/ackee";
+          ACKEE_USERNAME = "ackee";
+          ACKEE_PORT = "3000";
+        }
+      '';
+      description = ''
+        The configuration of Ackee is done through environment variables. The variables defined in
+        this settings will be set on the systemd service.
+
+        The available options can be found in
+        <link xlink:href="https://github.com/electerious/Ackee/blob/v${pkgs.ackee.version}/docs/Options.md">the documentation</link>.
+      '';
+    };
+
+    settingsFile = mkOption {
+      type = types.path;
+      example = "/etc/ackee/credentials";
+      description = ''
+        Environment file as defined in <citerefentry>
+        <refentrytitle>systemd.exec</refentrytitle><manvolnum>5</manvolnum>
+        </citerefentry>.
+
+        This file should be owned by root and not be readable by anyone but root.
+        The file complements the <option>settings</option> option and is supposed to contain
+        confidential settings like <literal>ACKEE_PASSWORD</literal>.
+
+        <programlisting>
+          # content of the environment file
+          ACKEE_PASSWORD=verysecretpassword
+        </programlisting>
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Warn on outdated mongodb
+    warnings = mkIf
+      (settings.ACKEE_MONGODB == "mongodb://localhost:27017/ackee"
+        && config.services.mongodb.enable == true
+        && versionOlder config.services.mongodb.package.version "4.0.6") [
+      "Ackee needs at least mongodb 4.0.6"
+    ];
+
+    systemd.services.ackee = {
+      description = "Ackee Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "networking.target" "mongodb.service" ];
```

Comment:
What if they aren't using a local `mongodb`?

Reply:
Then this would still wait until a local mongodb is started. If there is no local mongodb service, it will ignore it and start without waiting.
So yes, this is not optimal, but I think it’s not too bad either (matching `ACKEE_MONGODB` seems more brittle, especially as it can be provided in the `settingsFile`).
I’m happy to change that if there is a better way.

Reply:
Sounds good 👍